meson.build

```diff
@@ -1475,7 +1475,7 @@ endif
 macos_sdk_path = ''
 macos_sdk_version = '0.0'
 if darwin and macos_sdk_version_py.found()
-    macos_sdk_info = run_command(macos_sdk_version_py, check: true).stdout().split(',')
+    macos_sdk_info = run_command(python, macos_sdk_version_py, check: true).stdout().split(',')
```

Comment:
This is hardly the only python script though. I presume you explicitly had to change the python call since it was using the system one before. It's probably worth doing this for all of them to be consistent instead of having the possibility of having different python binaries run scripts.

Reply:
Updated. I hope I didn't miss something.

Reply:
There's also file2string

Reply:
msys2 builds are failing now...
```
"D:/a/_temp/msys64/clang64/bin/python.exe" "D:/a/_temp/msys64/clang64/bin/python.exe" "D:/a/mpv/mpv/TOOLS/matroska.py" "--generate-definitions" "ebml_defs.inc"
  File "D:/a/_temp/msys64/clang64/bin/python.exe", line 1

    MZx

SyntaxError: source code cannot contain null bytes
```

Reply:
That looks like an encoding issue. You could try specifying that it's utf-8. Not sure why this confused it.

Reply:
> There's also file2string

I don't see any usages of it. Am I missing something?

> That looks like an encoding issue. You could try specifying that it's utf-8. Not sure why this confused it.

Ok, I'll try it later.

Reply:
file2string is in the subdirectories. e.g. player/lua and player/javascript

Reply:
> file2string is in the subdirectories. e.g. player/lua and player/javascript

Added.

I've tried setting encoding via `# -*- coding: utf-8 -*-` but it's still failing. Or maybe it isn't the right way to do it?

Reply:
Try `# encoding=utf8`? I'm very confused why this would be even needed in the first place though.

Reply:
Doesn't work too -_-

Reply:
I did some testing of my own and basically just concluded it's something weird about whatever python mingw comes with. I guess let's just go back to the original patch and leave it as a mac-specific workaround. If you could leave a brief comment explaining, that would be cool.

Reply:
I tried to investigate it a bit more but no luck. I reverted the original patch and added commit message with explanation. Don't think any code comments are needed there or should I add them?

Reply:
Somebody is going to wonder why only this script is executed with python directly.

Reply:
Okay, this makes sense. Added.

Reply:
Thanks for all the trouble.